### ISSUE TITLE
Cannot preview SVG

### ISSUE THREAD

**Issue by davidbau** [PencilCode/pencilcode#186]
It seems that `data:image/svg...` image URLS are counted as tainted by the browsers (since it may hide url references within the svg), which means we can not screenshot SVG preview.

We should see if it is possible to preview an .svg file in a different way that can be screenshot.

**Comment by davidbau** [comment]
(Related to recently-closed bug #183)

**Comment by weihang7** [comment]
A quick Google search yields https://github.com/gabelerner/canvg. It seems to transfer the rendering to JavaScript to bypass the cross-domain restriction.

**Comment by davidbau** [comment]
Yes, but here's something simpler: I suspect if we use an inline <svg>
rather than an <img src="data:image/svg...">, then the browser won't have
the cross-domain problem because it will see there is not URL reference in
our SVG.

Note that html2canvas already includes svg rendering logic - a different
implementation than canvg I believe.   But it doesn't know how to get
around an svg data-url.

**Comment by weihang7** [comment]
Darn. I misunderstood and thought it was already an inline svg, and was confused as to why the browser rejected it...

**Comment by xinan** [comment]
I could try to play around the svg version of `html2canvas` to see how it goes.

**Comment by davidbau** [comment]
Assigning to @xinan - feel free to bounce it back if you don't have time.
